components/view_image/view_image.tsx

```diff
@@ -21,170 +24,203 @@ import PopoverBar from './popover_bar';
 
 const KeyCodes = Constants.KeyCodes;
 
-export default class ViewImageModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * The post the files are attached to
-         */
-        postId: PropTypes.string,
-
-        /**
-         * The post the files are attached to
-         */
-        post: PropTypes.object,
-
-        /**
-         * Set whether to show this modal or not
-         */
-        show: PropTypes.bool.isRequired,
-
-        /**
-         * Function to call when this modal is dismissed
-         **/
-        onModalDismissed: PropTypes.func.isRequired,
-
-        /**
-         * List of FileInfo to view
-         **/
-        fileInfos: PropTypes.arrayOf(PropTypes.object).isRequired,
-
-        /**
-         * The index number of starting image
-         **/
-        startIndex: PropTypes.number,
-
-        canDownloadFiles: PropTypes.bool,
-        enablePublicLink: PropTypes.bool,
-        pluginFilePreviewComponents: PropTypes.arrayOf(PropTypes.object),
-    };
+export type Props = {
+
+    /**
+ * The post the files are attached to
+ */
+    postId?: string;
+
+    /**
+ * The post the files are attached to
+ */
+    post?: Post;
+
+    /**
+ * Set whether to show this modal or not
+ */
+    show?: boolean;
+
+    /**
+ * Function to call when this modal is dismissed
+ **/
+    onModalDismissed?: () => void;
+
+    /**
+ * List of FileInfo to view
+ **/
+    fileInfos?: FileInfo[];
+
+    /**
+ * The index number of starting image
+ **/
+    startIndex?: number;
+
+    canDownloadFiles?: boolean;
+    enablePublicLink?: boolean;
+    pluginFilePreviewComponents?: unknown[];
+};
+
+type State = {
+    imageIndex?: number;
+    imageHeight: string;
+    loaded: boolean[];
+    progress: number[];
+    showCloseBtn: boolean;
+    showZoomControls: boolean;
+    scale: number[];
+    prevFileInfosCount: number;
+}
 
+export default class ViewImageModal extends React.PureComponent<Props, State> {
     static defaultProps = {
         show: false,
         fileInfos: [],
         startIndex: 0,
         pluginFilePreviewComponents: [],
         post: {}, // Needed to avoid proptypes console errors for cases like channel header, which doesn't have a proper value
     };
+    videoRef: React.RefObject<unknown>;
```

Comment:
@Willyfrog I don't understand what type should I give to the `RefObject.`

Reply:
not a 100% sure, but shouldn't it be `React.RefObject<HTMLVideoElement>`?